### PR TITLE
added profile command & helper functions for it

### PR DESCRIPTION
**Pull request by TogarashiPepper**
okay, try number 3, lets hope i didnt mess it up again

todo:
- ~~rebase~~
- ~~fix eslint errors~~
- ~~use `export default async function` instead of declaring then exporting~~

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 node_modules/
 config.json
+config.js
```
Comment (itsUrcute) [inline .gitignore:3]: no `config.js` files being used. should be removed.

File: src/commands/general/profile.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    {
+    	name: 'query',
+        type: 'STRING',
+        description: 'the username of the user to fetch',
+        required: true,
+    }
+  ],
+  permissions: [],
+  async execute(interaction) {
+    if(interaction.isCommand()) {
```
Comment (itsUrcute) [inline src/commands/general/profile.ts:18]: the command handler only passes `CommandInteraction` objects into `commands` so this check is redundant.

File: .gitignore (hunk 1 above)
Comment (TogarashiPepper) [reply .gitignore:3]: I added this before the switch was made, i will remove it

File: src/commands/general/profile.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import responses from "../../constants/responses";
+import type { Command } from "../../types/Command";
+import fetchUser from "../../util/fetchUser";
+import profileEmbed from "../../util/profileEmbed"; 
+export const command: Command = {
+  name: "profile",
+  description: "get the binarysearch profile of a user",
+  options: [
+    {
+    	name: 'query',
```
Comment (itsUrcute) [inline src/commands/general/profile.ts:10]: could use a more descriptive arg name like user instead of query

Comment (TogarashiPepper) [reply src/commands/general/profile.ts:10]: yeah, il change that

**Review by itsUrcute — CHANGES_REQUESTED** [review]
several newlines missing at EOFs.
needs proper linting.
use the `.gitignore` and `tsconfig.json` from [here](https://github.com/binarysus/algo-chan/pull/8).

File: src/commands/general/profile.ts (hunk 2 above)
Comment (itsUrcute) [reply src/commands/general/profile.ts:18]: it is but It already does the check in the command handler [here](https://github.com/binarysus/algo-chan/blob/master/src/internals/commandHandler.ts#L104).

Comment (TogarashiPepper) [reply src/commands/general/profile.ts:18]: oh, my apologies

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 dist/
 node_modules/
 config.json
+dist/
```
Comment (cheeze2000) [inline .gitignore:4]: extraneous line

File: src/constants/bsUser.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    },
+    warningLevel: null,
+    invites: [],
+    activeRoomUniqueSlug: null,
+    solved: number[],
+    solvedGlobal: number[],
+    submitted: number[],
+    solvedToday: number,
+    isOnline: boolean,
+// eslint-disable-next-line semi
```
Comment (cheeze2000) [inline src/constants/bsUser.ts:130]: try to use `@typescript-eslint/semi` and turn off `semi` if the rules are clashing in the eslintrc

File: src/util/fetchUser.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import fetch from "node-fetch";
+import BSUser from "../constants/bsUser";
+import urls from "../constants/urls";
+
+async function fetchUser(username: string): Promise<null | BSUser> {
+  const url = urls.BINARYSEARCH_API.replace("USERNAME", username);
+  const data = await fetch(url).then(res => res.json());
+
+  if ("user" in data) return data;
```
Comment (cheeze2000) [inline src/util/fetchUser.ts:9]: the `"profile"` field in the payload is not that important, should just `return data.user`

**Review by cheeze2000 — CHANGES_REQUESTED** [review]

File: src/util/fetchUser.ts (hunk 6 above)
Comment (TogarashiPepper) [reply src/util/fetchUser.ts:9]: got it

File: src/constants/bsUser.ts (hunk 5 above)
Comment (TogarashiPepper) [reply src/constants/bsUser.ts:130]: yeah sure

Comment (TogarashiPepper) [reply src/constants/bsUser.ts:130]: but also it's good to have the semi rule for other files, no?

File: .gitignore (hunk 4 above)
Comment (TogarashiPepper) [reply .gitignore]: should be fixed

File: src/constants/bsUser.ts (hunk 5 above)
Comment (cheeze2000) [reply src/constants/bsUser.ts:130]: you aren't linting anything else apart from ts files

File: src/util/fetchUser.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import fetch from "node-fetch";
```
Comment (cheeze2000) [inline src/util/fetchUser.ts:1]: please reflect new packages in `package.json`

**Review by cheeze2000 — CHANGES_REQUESTED** [review]

File: src/constants/bsUser.ts (hunk 5 above)
Comment (TogarashiPepper) [reply src/constants/bsUser.ts]: i turned of the semi rule as you suggested

File: src/commands/general/profile.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import responses from "../../constants/responses";
+import type { Command } from "../../types/Command";
+import fetchUser from "../../util/fetchUser";
```
Comment (itsUrcute) [inline src/commands/general/profile.ts:3]: why use `../../util/fetchUser` as a middleman instead of using `node-fetch` directly?

**Review by itsUrcute — COMMENTED** [review]
looks good :+1: : )

**Review by itsUrcute — APPROVED** [review]

**Review by itsUrcute — CHANGES_REQUESTED** [review]
please pull the latest changes

**Review by itsUrcute — APPROVED** [review]

**Pull request merged**